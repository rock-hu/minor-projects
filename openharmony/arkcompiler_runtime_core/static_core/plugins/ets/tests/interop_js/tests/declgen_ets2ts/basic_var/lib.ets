/**
 * Copyright (c) 2025 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export let arr = [1, 2, 3]

export let w = new BigInt(123)

export let w1 = new Boolean(true)

export let w2 = new String("")

export let w3 = new Number(123)

export let w4 = new Byte()

export let d2 = new Date()

export let ab = new ArrayBuffer(128)

export const a: number = 1
export const b: byte = 1
export const c: short = 1
export const d: int = 1
export const e: long = 1
export const f: float = 1
export const g: double = 1
export const h: char = "1"
export const i: boolean = true
export const j: string = "1"
export const k: bigint = 1n

export const a1: Number = new Number()
export const b1: Byte = new Byte()
export const c1: Short = new Short()
export const d1: Int = new Int()
export const e1: Long = new Long()
export const f1: Float = new Float()
export const g1: Double = new Double()
export const h1: Char = new Char()
export const i1: Boolean = new Boolean()
export const j1: String = new String()
export const k1: BigInt = new BigInt()

export let s : string = "Hi"

export type Greeting = "Hello"
export let a2: "Hello" = "Hello"
export let b2: Greeting = "Hello"