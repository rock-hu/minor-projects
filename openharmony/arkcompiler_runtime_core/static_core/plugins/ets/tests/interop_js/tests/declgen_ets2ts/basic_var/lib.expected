/*
 * Copyright (c) 2025 Huawei Device Co., Ltd.
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
export declare let arr: Array<number>;
export declare let w: bigint;
export declare let w1: boolean;
export declare let w2: string;
export declare let w3: number;
export declare let w4: number;
export declare let d2: Date;
export declare const a: number;
export declare const b: number;
export declare const c: number;
export declare const d: number;
export declare const e: number;
export declare const f: number;
export declare const g: number;
export declare const h: string;
export declare const i: boolean;
export declare const j: string;
export declare const k: bigint;
export declare const a1: number;
export declare const b1: number;
export declare const c1: number;
export declare const d1: number;
export declare const e1: number;
export declare const f1: number;
export declare const g1: number;
export declare const h1: string;
export declare const i1: boolean;
export declare const j1: string;
export declare const k1: bigint;
export declare let s: string;
export declare let a2: "Hello";
export declare let b2: Greeting;
export declare let a3: Record<string, string>;
export type Greeting = "Hello";
